Makefile

```diff
@@ -27,14 +27,34 @@ lint:
 unit_tests:
 	go test -race $$(go list ./... | grep -v integration_tests)
 
-integration_tests: build start_mongo
+integration_tests: build start_postgres init_test_db
 	go test -race -v ./integration_tests
 
-start_mongo:
-	./mongo.sh start
+start_postgres:
+	@if [ -z $$(docker ps -aqf name=router-postgres-test-db) ]; then \
+		docker run \
+			--name router-postgres-test-db \
+			-e POSTGRES_HOST_AUTH_METHOD=trust \
+			-d \
+			-p 5432:5432 \
+			--user 'postgres' \
+			--health-cmd 'pg_isready' \
+			--health-start-period 5s \
+			postgres:14; \
+		echo Waiting for postgres to be up; \
+		until [ "`docker inspect -f '{{.State.Health.Status}}' router-postgres-test-db`" = "healthy" ]; do \
```

Comment:
optional: if the docker or the container fails then this could get stuck for a long time (6h on GitHub Actions), so could so with `for _ in $(seq 30)`

Reply:
sorry that was meant to be sent along with the review rather than as a one-off (I hit the wrong button)